src/os_macosx.m

```diff
@@ -421,18 +418,14 @@ - (void) sound:(NSSound *)sound didFinishPlaying:(BOOL)flag
 	    delete_sound_callback(_callback);
 	    _callback = NULL;
 	}
-	[sounds_list removeObjectForKey:[NSNumber numberWithLong:_sound_id]];
     }
-    // Release itself. Do that here instead of earlier because NSSound only
-    // holds weak reference to this object.
-    [self release];
 }
 @end
 
     void
 process_cfrunloop(void)
 {
-    if (sounds_list != nil && [sounds_list count] > 0)
+    if (sounds_list != nil && [sounds_list count])
```

Comment:
FWIW I don't particularly like using `if (<int>) {}`. It loses the semantic meaning of what you are trying to do and relying on an implicit behavior. As an optimization it also wouldn't gain much as the compiler understands how to optimize `!= 0` etc away.